### PR TITLE
Pretrained subclasses nn.Module now

### PR DESCRIPTION
**Pull request by Gastron**
Addresses issue #987.

I've tried to make some sane choice for the forward() method of each subclass. Some models cannot pass gradients through their decode procedure in any meaningful way - is that unexpected for a forward() method? Should we only pick methods that pass gradients?

### CODE REVIEW

**Comment by aheba** [comment]
Hi @Gastron , 
let me fork this PR and work on an the proposed solution using import lib and external injection of the forward function
